edu.harvard.i2b2.ontology/src/core/edu/harvard/i2b2/ontology/dao/ConceptDao.java

```diff
@@ -284,8 +284,11 @@ else if (childrenType.getType().equals("all")){
 		if(childrenType.isSynonyms() == false)
 			synonym = " and c_synonym_cd = 'N'";
 
-		String sql = "select " + parameters +" from " + metadataSchema+tableName  + " where c_fullname like ? " + (!dbInfo.getDb_serverType().toUpperCase().equals("POSTGRESQL") ? "{ESCAPE '?'}" : "" ) + " and c_hlevel = ? "; 
-		sql = sql + hidden + synonym + " order by upper(c_name) ";
+        // get all children if the numLevel is less then zero
+        int numLevel = childrenType.getNumLevel();
+        String sql = "select " + parameters + " from " + metadataSchema + tableName + " where c_fullname like ? " + (!dbInfo.getDb_serverType().toUpperCase().equals("POSTGRESQL") ? "{ESCAPE '?'}" : "");
```

Comment:
Shouldn't c_hlevel = ? be in 289? You have > ? now.

Reply:
No, c_hlevel should not be ***c_hlevel = ?***.   The **c_hlevel > level** for ***all levels*** from the parent. The ***c_hlevel*** is the range from the **level** to the **level + numLevel** for the number of levels from the parent >= 0.

Reply:
Kevin, thanks for the explanation. We'll discuss this at our meeting this afternoon.

Can you write some documentation on how the new API works? We'll want to add it to this wiki page: https://community.i2b2.org/wiki/display/ServerellsMessagingHome/get_children

Reply:
Hi Jeff.  Sure.  I'll read up on the existing documentation and write up some documentation on the new parameter.